File: src/main/javabak/tile_entities/capabilities/EmptyEnergyStorage.java
package redcrafter07.processed.block.tile_entities.capabilities;

public class EmptyEnergyStorage implements IEnergyStorageModifiable {
    public static final IEnergyStorageModifiable INSTANCE = new EmptyEnergyStorage();

    @Override
    public void setEnergyStored(int energy) {}
    @Override
    public void setMaxEnergyStored(int maxEnergy) {}

    @Override
    public int receiveEnergy(int toReceive, boolean simulate) {
        return 0;
    }

    @Override
    public int extractEnergy(int toExtract, boolean simulate) {
        return 0;
    }

    @Override
    public int getEnergyStored() {
        return 0;
    }

    @Override
    public int getMaxEnergyStored() {
        return 0;
    }

    @Override
    public boolean canExtract() {
        return false;
    }

    @Override
    public boolean canReceive() {
        return false;
    }
}

File: src/main/kotlin/redcrafter07/processed/block/tile_entities/capabilities/EmptyEnergyStorage.kt
package redcrafter07.processed.block.tile_entities.capabilities

object EmptyEnergyStorage : EnergyStorageModifiable {
    override fun setEnergyStored(energy: Int) = Unit
    override fun setMaxEnergyStored(maxEnergy: Int) = Unit
    override fun receiveEnergy(p0: Int, p1: Boolean): Int = 0
    override fun extractEnergy(p0: Int, p1: Boolean): Int = 0
    override fun getEnergyStored(): Int = 0
    override fun getMaxEnergyStored(): Int = 0
    override fun canExtract(): Boolean = false
    override fun canReceive(): Boolean = false
}